### PR TITLE
docs(cdaas): DOC-721 Add CDaaS quickstart; CDAAS-1604 add kubectl RNA

### PR DESCRIPTION
**Pull request by aimeeu**
Resolves Jira: [CDAAS-1604] [DOC-721] [CDAAS-2365]

1. Add Quickstart (enhance and add more screenshots)
1. Install a Remote Network Agent page: add CLI and kubectl RNA install methods to 
1. Change Tasks to Guides (section name; no directory changes)
1. I went down a HUGE RABBIT HOLE. With the addition of the Quickstart, much of the material in the CLI page (setup/cli) was redundant. SO I moved the CLI install to a new page in the Guides section and added a Deploy Your Own App page to the Get Started section.  
1. Delete Register page (redundant). Move pertinent info to Quickstart.
1. Add new `youtube-enhanced` shortcode that takes in width/height percents and uses non-tracking URL; update CDaaS landing page to use the new shortcode.

Deploy previews:
- Quickstart https://deploy-preview-1986--armory-docs.netlify.app/cd-as-a-service/setup/quickstart/
- Deploy Your Own App  https://deploy-preview-1986--armory-docs.netlify.app/cd-as-a-service/setup/deploy-your-app/
- CLI Install  https://deploy-preview-1986--armory-docs.netlify.app/cd-as-a-service/tasks/cli/
- Agent install https://deploy-preview-1986--armory-docs.netlify.app/cd-as-a-service/tasks/networking/install-agent/    -> added UI wizard and kubectl instructions (note that the UI uses both **Cluster Name** and **Agent Identifier** to refer to RNA name)

**Note:  links in the UI must be updated when this is merged! See DOC-721 for specific UI screen**

Quickstart content provided by Dan Peach https://github.com/armory-io/cdaas-examples/tree/main/hello-armory

[CDAAS-1604]: https://armory.atlassian.net/browse/CDAAS-1604?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DOC-721]: https://armory.atlassian.net/browse/DOC-721?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[CDAAS-2365]: https://armory.atlassian.net/browse/CDAAS-2365?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: content/en/cd-as-a-service/setup/quickstart.md
Comment (parth-srivastav) [inline content/en/cd-as-a-service/setup/quickstart.md:66]: maybe we should not call out 2 namespaces, If this sample app scales, this becomes outdated immediately

Comment (aimeeu) [reply content/en/cd-as-a-service/setup/quickstart.md:66]: @parth-srivastav Unfortunately the second the sample app changes (or the UI) most of the quickstart becomes outdated - more namespaces, another deployment strategy, a webhook... 
After this PR merges, I plan to update the hello-armory repo README to point to the published quickstart... so hopefully engineers will update docs if they update the hello-armory example.

Comment (aimeeu) [reply content/en/cd-as-a-service/setup/quickstart.md:66]: I will also update the existing tutorials to deploy the potato-facts app.

File: content/en/cd-as-a-service/setup/quickstart.md
Comment (parth-srivastav) [inline content/en/cd-as-a-service/setup/quickstart.md]: its 75/25, I checked the strategy file. 
or we should explicitly say, 75% traffic to old version and 25% traffic to new version

File: content/en/cd-as-a-service/setup/quickstart.md
Comment (parth-srivastav) [inline content/en/cd-as-a-service/setup/quickstart.md]:
```suggestion
Use the link provided by the CLI to navigate to your deployment in the [CD-as-a-Service Console](https://console.cloud.armory.io/deployments). Once the `staging` deployment has completed, click **Approve** to allow the `prod` deployment to begin.
```

**Review by parth-srivastav — COMMENTED** [review]
Small clarifications on text. Content looks great!

File: content/en/cd-as-a-service/setup/quickstart.md
Comment (aimeeu) [reply content/en/cd-as-a-service/setup/quickstart.md]: Thanks for pointing that out. The original said 20/80...

File: content/en/cd-as-a-service/setup/deploy-your-app.md
Comment (parth-srivastav) [inline content/en/cd-as-a-service/setup/deploy-your-app.md]: tenant selection occurs only for users with more than 1 tenants.

If you have more than one tenants, Select the tenant you want to access

File: content/en/cd-as-a-service/setup/deploy-your-app.md
Comment (parth-srivastav) [inline content/en/cd-as-a-service/setup/deploy-your-app.md]: The `deploy start` command starts your deployment and returns a Deployment ID and a link to your deployment. You can navigate to the deployment details page in CD-as-a-Service Console to see your deployment execution. 

The command itself just starts the deployment, and returns ID and URL. The deployment runs according to strategy.

File: content/en/cd-as-a-service/tasks/cli.md
Comment (parth-srivastav) [inline content/en/cd-as-a-service/tasks/cli.md]: Its possible to install the CLI on windows by downloading the CLI from the releases page. 
https://github.com/armory-io/armory-cli/releases/tag/v1.14.5

File: content/en/cd-as-a-service/tasks/cli.md
Comment (parth-srivastav) [inline content/en/cd-as-a-service/tasks/cli.md]: We can use the Tab pane like you have used below for Mac & Linux installation

File: content/en/cd-as-a-service/tasks/networking/install-agent.md
Comment (parth-srivastav) [inline content/en/cd-as-a-service/tasks/networking/install-agent.md]: I'd propose adding how to create credential steps right here, like we have CLI installation steps. It can be very high level

1. got this URL https://console.cloud.armory.io/configuration/credentials/new
2. Create a credential with "Remote Network Agent" permissions

File: content/en/cd-as-a-service/setup/quickstart.md
Comment (parth-srivastav) [inline content/en/cd-as-a-service/setup/quickstart.md:26]: I'll also propose adding the text below, for users who are not ready to install kind, or are just exploring docs before sign up.
-  if you do not have a cluster and still wish to tour the product, You can sign up and select "Self-Guided Browser Tour". We spin a short-lived Kubernetes cluster to deploy the sample application.

File: content/en/cd-as-a-service/_index.md
Comment (parth-srivastav) [inline content/en/cd-as-a-service/_index.md]:
```suggestion
* {{< linkWithTitle "cd-as-a-service/setup/gh-action.md" >}} guide walks you through integrating Armory CD-as-a-Service's GitHub Action into your workflow.
```

File: content/en/cd-as-a-service/setup/deploy-your-app.md
Comment (aimeeu) [reply content/en/cd-as-a-service/setup/deploy-your-app.md]: Decided to remove that line. For this guide, probably all the readers will only have one tenant.

File: content/en/cd-as-a-service/setup/deploy-your-app.md
Comment (aimeeu) [reply content/en/cd-as-a-service/setup/deploy-your-app.md]: Yes... and the strategy used for this guide has a manual approval for prod deployment.  Clarified the content in the _Deploy first version_ section.

File: content/en/cd-as-a-service/tasks/networking/install-agent.md
Comment (aimeeu) [reply content/en/cd-as-a-service/tasks/networking/install-agent.md]: Added the reusable code from the Create Client Credentials page. Surrounded with `<details>` so user can expand for instructions.

**Review by parth-srivastav — APPROVED** [review]

**Pull request merged**
